azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java

```diff
@@ -142,6 +152,36 @@ protected void config(WebAppConfiguration configuration) throws MojoFailureExcep
             }
         } while (!confirmConfiguration(result));
         Log.info(SAVING_TO_POM);
+        // legacy code: to distinguish the non-value changes of javaVersion and webContainer: eg: jre8 to Java 8, jre8 -> Java SE
+        if (Objects.nonNull(configuration) && Objects.nonNull(result.getOs()) && Objects.nonNull(this.getRuntime())) {
+            switch (result.getOs()) {
+                case Linux:
+                    if (!StringUtils.equals(
+                            RuntimeStackUtils.getJavaVersionFromRuntimeStack(result.getRuntimeStack()),
+                            this.getRuntime().getJavaVersionRaw()) ||
+                        !StringUtils.equals(
+                                RuntimeStackUtils.getWebContainerFromRuntimeStack(result.getRuntimeStack()),
+                                this.getRuntime().getWebContainerRaw())) {
+                        FieldUtils.writeField(configuration, "runtimeStack", null, true);
```

Comment:
add set method in `WebAppConfiguration`?

Reply:
no need since it breaks the builder integration, this piece of code will be deleted in future refactor, since we need a real model representing user's configuration other than WebConfiguration which is mainly used in deploy goal.